src/Domain/DonationRepository.php

```diff
@@ -177,7 +171,7 @@ public function buildFromApiRequest(DonationCreate $donationData): Donation
 
         $donation = new Donation();
         $donation->setPsp($donationData->psp);
-        $donation->setPaymentMethodType($donationData->paymentMethodType);
+        $donation->setPaymentMethodType($donationData->paymentMethodType->name);
```

Comment:
If we move the enum into the domain model (or even if we don't) we could share it between there and the HTTP model. Then this can be simplified to

```suggestion
        $donation->setPaymentMethodType($donationData->paymentMethodType);
```

I think that's a lot better. The enum has to be string-backed, and doctrine can be configured to persist it with an annotation like we have for the status field:

https://github.com/thebiggive/matchbot/blob/38dc7445c67ee1ccc763476046477227d56ef97a/src/Domain/Donation.php#L146

It doesn't mean it has to be an enum in MySQL.